### PR TITLE
PR to fix #396 and #404

### LINKED ISSUE #396: Fix visualization for single-class mode
We would like to apply a threshold on outputs to get accurate visualization. 

Solution: apply threshold after line https://github.com/NRCan/geo-deep-learning/blob/8678cd900f12d2ded72395131b804c3babd74474/utils/visualization.py#L130

### PR DESCRIPTION
**Pull request by ms888ekb**
fixes #396, #404

Fixed vis() function
Fixed subset_multiband_vrt() function
Minor syntax fixes

### CODE REVIEW

**Review by remtav — COMMENTED** [review]
Thanks for addressing this.
Following our development workflow, I'd recommend splitting this PR into 2 separate PRs: 
1. bug fix for #396 
2. bug fix for subset_multiband_vrt()
For this second PR, first step would be to open an issue to document bug and provide minimal code to detect bug. For example, can you figure out why our [unit test](https://github.com/NRCan/geo-deep-learning/blob/develop/tests/dataset/test_aoi.py#L37) did not pick up on this bug? Can we modify this test to better cover this feature?

**Comment by ms888ekb** [comment]
> 2\. bug fix for subset_multiband_vrt()
>     For this second PR, first step would be to open an issue to document bug and provide minimal code to detect bug. For example, can you figure out why our [unit test](https://github.com/NRCan/geo-deep-learning/blob/develop/tests/dataset/test_aoi.py#L37) did not pick up on this bug? Can we modify this test to better cover this feature?

This issue already exists:
https://github.com/NRCan/geo-deep-learning/issues/404

The test does not cover a case when we subset only one band (it covers only 2-band subsetting).
The testing unit is: test_multiband_input_band_selection() with the inputs:
bands_request = [2, 1]
If you change it to:
bands_request = [1]
it fails.

**Review by remtav — APPROVED** [review]

**Comment by ms888ekb** [comment]
extended a unit test for the band selection function

File: tests/dataset/test_aoi.py [hunk 1]
```diff
@@ -34,12 +35,15 @@ def test_multiband_input(self):
         aoi.close_raster()
         assert aoi.raster_closed
 
-    def test_multiband_input_band_selection(self):
+    @pytest.fixture(
+        params=[[2, 1], [1], [2], [3], [1, 3]]
+    )
+    def test_multiband_input_band_selection(self, request: SubRequest) -> None:
```
Comment (remtav) [inline tests/dataset/test_aoi.py:41]: Cool!
@ms888ekb are you able to use Pycharm's debugger with the "test_multiband_input_band_selection" pytest fixture you added? I can't get it to work even though it works when debugging a very similarly structured test in torchgeo: https://github.com/microsoft/torchgeo/blob/main/tests/samplers/test_single.py#L94

Also, when I simply run tests from test_aoi.py, I get a total of 30 tests when the fixture is added, whereas I get total=31 without. It seems to be skipping this specific test when adding fixtures. I played around a bit to try and figure out why torchgeo's test works, but don't understand...

Comment (remtav) [reply tests/dataset/test_aoi.py:41]: FYI, this is what I get when trying to run this specific test:
image

Comment (remtav) [reply tests/dataset/test_aoi.py:41]: This works:
```
    @pytest.fixture(
        params=[[2, 1], [1], [2], [3], [1, 3]]
    )
    def bands_request(self, request: SubRequest) -> List:
        return request.param

    def test_multiband_input_band_selection(self, bands_request):
        """Tests reading a multiband raster as input with band selection"""
        extract_archive(src="tests/data/spacenet.zip")
        data = read_csv("tests/tiling/tiling_segmentation_binary-multiband_ci.csv")
        row = data[0]
        aoi = AOI(raster=row['tif'], label=row['gpkg'], split=row['split'], raster_bands_request=bands_request)
        ...
```

Comment (ms888ekb) [reply tests/dataset/test_aoi.py]: Submitted the fix

**Review by remtav — APPROVED** [review]

**Review by CharlesAuthier — APPROVED** [review]

**Pull request merged**